src/processor/Package.ts

```diff
@@ -228,4 +230,44 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  private suppressUrlAssignmentOnExtensions(): void {
+    // Loop over all profiles and extensions, removing assignment rules on inline extensions
+    [...this.profiles, ...this.extensions].forEach(sd => {
+      const rulesToRemove: number[] = [];
+      sd.rules.forEach(rule => {
+        if (rule instanceof ExportableContainsRule && rule.path.endsWith('extension')) {
+          rule.items.forEach(item => {
+            const assignmentRuleIdx = sd.rules.findIndex(
+              other =>
+                other.path === `${rule.path}[${item.name}].url` &&
+                other instanceof ExportableFixedValueRule &&
+                other.fixedValue === item.name
+            );
+            if (assignmentRuleIdx >= 0) {
+              rulesToRemove.push(assignmentRuleIdx);
+            }
+          });
+        }
+      });
+      pullAt(sd.rules, rulesToRemove);
+    });
+    // We must know the configuration to determine if a rule fixing "url" matches the url that SUSHI will assume
+    if (this.configuration?.config?.canonical) {
+      this.extensions.forEach(extension => {
+        const rulesToRemove: number[] = [];
+        extension.rules.forEach((rule, i) => {
+          if (
+            rule instanceof ExportableFixedValueRule &&
+            rule.path === 'url' &&
+            rule.fixedValue ===
+              `${this.configuration.config.canonical}/StructureDefinition/${extension.id}`
```

Comment:
This will usually be right, but it will be wrong if/when the author overrides the URL with one that is outside the canonical.  In this case, we would actually have an `ExportableCaretValueRule` for that (e.g., `* ^url = "http://foo.org/MyExtension"`).

If we're not going to try to bring the SD down to the optimization code (as discussed in my previous comment) we probably need to accommodate the use case where there might be a caret rule overriding the URL.

Reply:
So as discussed above, my understanding is that such a rule is impossible because `url` is in `IGNORED_PROPERTIES`:
```
const IGNORED_PROPERTIES = [
  'resourceType',
  'id',
  'url', // NOTE: For now, we assume URL matches canonical and is generated
  'name',
  'title',
  'description',
  'fhirVersion',
  /*'mapping', NOTE: mapping should be ignored when a mapping extractor is implemented*/
  'baseDefinition',
  'derivation',
  'snapshot',
  'differential'
];
```

Reply:
Good point.  Although now that we support getting a canonical from the config, we should probably change that so `url` is not ignored anymore.  But that's outside the scope of this PR.